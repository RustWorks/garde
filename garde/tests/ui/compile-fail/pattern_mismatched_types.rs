static STR: &str = r"a|b";

#[derive(garde::Validate)]
struct Test<'a> {
    #[garde(pattern(STR))]
    field: &'a str,
}

fn main() {}
