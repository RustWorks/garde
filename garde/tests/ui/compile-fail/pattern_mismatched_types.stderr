error[E0308]: mismatched types
 --> tests/ui/compile-fail/pattern_mismatched_types.rs:5:21
  |
5 |     #[garde(pattern(STR))]
  |                     ^^^ expected `&Regex`, found `&&str`
  |
  = note: expected reference `&Regex`
             found reference `&&'static str`
